### PR TITLE
[SDK-453] Feature/precompile support

### PR DESCRIPTION
**Pull request by HarrisonHough**
## Description

-  this adds support for using the precompile endpoint to slightly increase avatar update/load times during avatar creation process.

## How to Test

- Import CAC package and run the sample scene. 
- switch categories  and wait  a few seconds (as precompile takes some time to complete
- then switch to one of the first 20 assets of that category
- In theory, these first 20 should generally load faster (0.5s -> 0.8s for outfits) compared to others (~1s -> 2s)

## Checklist

-   [ ] Tests written or updated for the changes.
-   [ ] Documentation is updated.
-   [ ] Changelog is updated.

[SDK-453]: https://ready-player-me.atlassian.net/browse/SDK-453?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Runtime/PartnerAssetsManager.cs [hunk 1]
```diff
@@ -25,6 +26,12 @@ public class PartnerAssetsManager : IDisposable
         private OutfitGender gender;
         private CancellationTokenSource ctxSource;
 
+        public PrecompileData GetPrecompileData(Category[] categories, int numberOfAssetsPerCategory)
+        {
+            var dictionary = categories.ToDictionary(category => category.ToString().PascalToCamelCase(), category => GetAssetsByCategory(category).ToArray().Take(numberOfAssetsPerCategory).ToArray());
```
[the remark this answers is not in the document]
Comment (rYuuk) [reply Runtime/PartnerAssetsManager.cs:31]: The reason being as you can see that not all type follows same case.

**Review by rYuuk — CHANGES_REQUESTED** [review]
Great job, everything looks fine and also shaves some time in loading avatar. I have a few concerns, please have a look.

File: Runtime/PartnerAssetsManager.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (HarrisonHough) [reply Runtime/PartnerAssetsManager.cs:31]: okay updated, I had to use some linq magic to get the keys :D

File: Runtime/PartnerAssetsManager.cs [hunk 2]
```diff
@@ -25,6 +26,12 @@ public class PartnerAssetsManager : IDisposable
...
+            return new PrecompileData { data = dictionary };
+        }
```
[the remark this answers is not in the document]
Comment (HarrisonHough) [reply Runtime/PartnerAssetsManager.cs:33]: updated

File: Runtime/WebRequests/AvatarAPIRequests.cs [hunk 3]
```diff
@@ -197,6 +192,22 @@ public async Task<byte[]> UpdateAvatar(string avatarId, AvatarProperties avatarP
...
+            var response = await authorizedRequest.SendRequest<Response>(
+                new RequestData
+                {
+                    Url = AvatarEndpoints.GetPrecompileEndpoint(avatarId, parameters),
+                    Method = HttpMethod.POST,
+                    Payload = json
+                },
+                ctx: ctx);
+
+            Debug.Log($"Precompile IS Success = {response.IsSuccess}");
```
[the remark this answers is not in the document]
Comment (HarrisonHough) [reply Runtime/WebRequests/AvatarAPIRequests.cs:207]: removed

File: Samples~.meta
Comment (rYuuk) [inline Samples~.meta:1]: oh it came again, I thought I deleted it 😓 
Thanks for deleting this.

File: Runtime/PartnerAssetsManager.cs [hunk 4]
```diff
@@ -132,5 +133,17 @@ public void DeleteAssets()
...
+        public PrecompileData GetPrecompileData(Category[] categories, int numberOfAssetsPerCategory)
+        {
+            var categoriesFromMap = CategoryHelper.PartnerCategoryMap
+                .Where(kvp => categories.Contains(kvp.Value))
+                .Select(kvp => kvp.Key)
+                .ToArray();
+
+            var dictionary = categoriesFromMap.ToDictionary(category => category, category => GetAssetsByCategory(CategoryHelper.PartnerCategoryMap[category]).ToArray().Take(numberOfAssetsPerCategory).ToArray());
+            
+            return new PrecompileData { data = dictionary };
```
Comment (rYuuk) [inline Runtime/PartnerAssetsManager.cs:146]: woah, linq magic 🪄

**Review by rYuuk — APPROVED** [review]
🟢 Thanks for making the changes 💯

**Pull request merged**